### PR TITLE
cgroup test update to check for memory_spread_page file

### PR DESCRIPTION
**Pull request by vccardenas**
#### Describe Bug or Feature
TestCgroupsHook.test_cgroup_cpuset_memory_spread_page intermittently fails due to missing cgroup dir cpuset.memory_spread_page.

#### Describe Your Change
Added a check if cpuset.memory_spread_page exists before proceeding with the rest of the test.

#### Attach Test Logs/Output
tests showing intermittent failure (2 out of 50):
[test_cgroup_cpuset_memory_spread_page_td-10_loop50_before.txt](https://github.com/PBSPro/pbspro/files/3387470/test_cgroup_cpuset_memory_spread_page_td-10_loop50_before.txt)

tests showing fixed issue:
[test_cgroup_cpuset_memory_spread_page_td-10_loop50_after1.txt](https://github.com/PBSPro/pbspro/files/3387472/test_cgroup_cpuset_memory_spread_page_td-10_loop50_after1.txt)

### CODE REVIEW

File: test/tests/functional/pbs_cgroups_hook.py [hunk 1]
```diff
@@ -819,7 +819,7 @@ def is_file(self, cpath, host):
         """
         Returns True if path exists otherwise false
         """
-        for _ in range(5):
+        for _ in range(20):
```
Comment (mike0042) [inline test/tests/functional/pbs_cgroups_hook.py]: Suggest you change this back to 5 attempts and add a time.sleep(0.5) after the check for rv. We don't need to continuously spin while waiting for the file to appear.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 2]
```diff
@@ -2193,20 +2193,18 @@ def test_cgroup_cpuset_memory_spread_page(self):
...
+        result = self.du.cat(hostname, filename=fn, sudo=True)
         self.assertEqual(result['rc'], 0)
         self.assertEqual(result['out'][0], '0')
         self.load_config(self.cfg5 % ('false', '', 'true', 'false',
                                       'true', self.swapctl))
-        self.server.expect(NODE, {'state': 'free'},
-                           id=self.nodes_list[0], interval=3, offset=10)
-        a = {'Resource_List.select': '1:ncpus=1:mem=100mb:host=%s' %
-             self.hosts_list[0]}
+        self.server.expect(NODE, {'state': 'free'}, nid, interval=3, offset=10)
```
Comment (mike0042) [inline test/tests/functional/pbs_cgroups_hook.py]: id=nid

File: test/tests/functional/pbs_cgroups_hook.py [hunk 3]
```diff
@@ -2215,10 +2213,9 @@ def test_cgroup_cpuset_memory_spread_page(self):
         self.server.status(JOB, ATTR_o, jid)
         o = j.attributes[ATTR_o]
         self.tempfile.append(o)
-        fn = self.get_cgroup_job_dir('cpuset', jid, self.hosts_list[0])
+        fn = self.get_cgroup_job_dir('cpuset', jid, hostname)
         fn = os.path.join(fn, spread_path)
-        result = self.du.cat(hostname=self.hosts_list[0],
-                             filename=fn, sudo=True)
+        result = self.du.cat(hostname, filename=fn, sudo=True)
```
Comment (mike0042) [inline test/tests/functional/pbs_cgroups_hook.py]: hostname=hostname
You might want to consider renaming your hostname variable to hn or hostn.

**Comment by vccardenas** [comment]
I have made changes per comments from @mike0042 . Please review once more.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 4]
```diff
@@ -2180,10 +2180,10 @@ def test_cgroup_cpuset_memory_spread_page(self):
         """
         self.load_config(self.cfg5 % ('false', '', 'true', 'false',
                                       'false', self.swapctl))
-        self.server.expect(NODE, {'state': 'free'},
-                           id=self.nodes_list[0], interval=3, offset=10)
-        a = {'Resource_List.select': '1:ncpus=1:mem=100mb:host=%s' %
-             self.hosts_list[0]}
+        id = self.nodes_list[0]
```
Comment (mike0042) [inline test/tests/functional/pbs_cgroups_hook.py]: I liked this better as nid.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 5]
```diff
@@ -2180,10 +2180,10 @@ def test_cgroup_cpuset_memory_spread_page(self):
...
+        self.server.expect(NODE, {'state': 'free'}, id, interval=3, offset=10)
```
Comment (mike0042) [inline test/tests/functional/pbs_cgroups_hook.py]: Notice that the previous version specified "id=" to eliminate issues with the parameter order. To retain this, you want to specify "id=nid" here.

File: test/tests/functional/pbs_cgroups_hook.py [hunk 6]
```diff
@@ -2193,20 +2193,18 @@ def test_cgroup_cpuset_memory_spread_page(self):
...
         spread_path = 'cpuset.memory_spread_page'
-        fn = self.get_cgroup_job_dir('cpuset', jid, self.hosts_list[0])
+        fn = self.get_cgroup_job_dir('cpuset', jid, hostn)
         if self.noprefix:
             spread_path = 'memory_spread_page'
         fn = os.path.join(fn, spread_path)
-        result = self.du.cat(hostname=self.hosts_list[0],
-                             filename=fn, sudo=True)
+        self.assertTrue(self.is_file(fn, hostn))
+        result = self.du.cat(hostn, filename=fn, sudo=True)
```
Comment (mike0042) [inline test/tests/functional/pbs_cgroups_hook.py]: Here is another case where the old version used "hostname=", so you should use "hostname=hostn".

File: test/tests/functional/pbs_cgroups_hook.py [hunk 7]
```diff
@@ -2193,20 +2193,18 @@ def test_cgroup_cpuset_memory_spread_page(self):
...
         self.assertEqual(result['rc'], 0)
         self.assertEqual(result['out'][0], '0')
         self.load_config(self.cfg5 % ('false', '', 'true', 'false',
                                       'true', self.swapctl))
-        self.server.expect(NODE, {'state': 'free'},
-                           id=self.nodes_list[0], interval=3, offset=10)
-        a = {'Resource_List.select': '1:ncpus=1:mem=100mb:host=%s' %
-             self.hosts_list[0]}
+        self.server.expect(NODE, {'state': 'free'}, id, interval=3, offset=10)
```
Comment (mike0042) [inline test/tests/functional/pbs_cgroups_hook.py]: id=nid

File: test/tests/functional/pbs_cgroups_hook.py [hunk 8]
```diff
@@ -2215,10 +2213,9 @@ def test_cgroup_cpuset_memory_spread_page(self):
         self.server.status(JOB, ATTR_o, jid)
         o = j.attributes[ATTR_o]
         self.tempfile.append(o)
-        fn = self.get_cgroup_job_dir('cpuset', jid, self.hosts_list[0])
+        fn = self.get_cgroup_job_dir('cpuset', jid, hostn)
         fn = os.path.join(fn, spread_path)
-        result = self.du.cat(hostname=self.hosts_list[0],
-                             filename=fn, sudo=True)
+        result = self.du.cat(hostn, filename=fn, sudo=True)
```
Comment (mike0042) [inline test/tests/functional/pbs_cgroups_hook.py]: hostname=hostn

File: test/tests/functional/pbs_cgroups_hook.py (hunk 5 above)
Comment (mike0042) [reply test/tests/functional/pbs_cgroups_hook.py]: Look at the side-by-side diff to see what I mean.

**Pull request merged**